### PR TITLE
Conceptual documentation for extensions.

### PR DESCRIPTION
**Pull request by sptramer**

### CODE REVIEW

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[latest/docs-ref-conceptual/TOC.yml](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |
[latest/docs-ref-conceptual/azure-cli-extensions.md](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/azure-cli-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C30%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201710302305535708-536%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=0MHjXjb5n%2BHtRKC%2FHhSfOyv8eqSa8GDAKDbL5K7VbcU%3D&st=2017-10-30T23%3A02%3A58Z&se=2017-11-30T23%3A07%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[latest/docs-ref-conceptual/TOC.yml](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |
[latest/docs-ref-conceptual/azure-cli-extensions.md](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/azure-cli-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C30%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201710302310007770-536%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=6DMEll9L%2BwyJTVus5dsJPHlnw50Ms%2ByhP3dmRtu6sd0%3D&st=2017-10-30T23%3A07%3A05Z&se=2017-11-30T23%3A12%3A05Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[latest/docs-ref-conceptual/TOC.yml](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |
[latest/docs-ref-conceptual/azure-cli-extensions.md](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/azure-cli-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C30%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201710302315500695-536%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=XIdD5jJ28iRG8%2FMJQDsYaWePkqezb2%2BTLUvBVrM0ZJs%3D&st=2017-10-30T23%3A13%3A10Z&se=2017-11-30T23%3A18%3A10Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: latest/docs-ref-conceptual/azure-cli-extensions.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+Once you have found an extension to install, use `az extension add` to get it. If the extension is an official Microsoft extension listed in `az extension list-available`, you can install the extension
+by name.
+
+```azurecli
+az extension add --name <extension-name>
+```
+
+If the extension is from an external resource or you have a direct link to it, you can provide the source URL or local path. This _must_ be a compiled Python wheel file.
+download.
```
Comment (derekbekoe) [inline latest/docs-ref-conceptual/azure-cli-extensions.md]: Typo? Can remove `download.`?

File: latest/docs-ref-conceptual/azure-cli-extensions.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Updating extensions
+
+Extensions can only be updated by name:
+
+```azurecli
+az extension update --name <extension-name>
+```
+
+If an extension name cannot be resolved by the CLI for whatever reason, reinstall the extension to update it. There is also the possibility that the extension was moved out of preview and became a built-in command for the CLI. You should always update the CLI before updating extensions, although there is not a guarantee extensions will continue to work with later CLI versions.
```
Comment (derekbekoe) [inline latest/docs-ref-conceptual/azure-cli-extensions.md]: I wouldn't want to enforce this statement "You should always update the CLI before updating extensions".
This makes it sound like it is a MUST.

File: latest/docs-ref-conceptual/azure-cli-extensions.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## Uninstalling extensions
+
+If you no longer need an extension, it can be removed with `az extension remove`,
```
Comment (derekbekoe) [inline latest/docs-ref-conceptual/azure-cli-extensions.md]: full stop?

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[latest/docs-ref-conceptual/TOC.yml](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |
[latest/docs-ref-conceptual/azure-cli-extensions.md](https://github.com/sptramer/azure-docs-cli-python/blob/extensions/reference/latest/docs-ref-conceptual/azure-cli-extensions.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/azure-cli-latest/azure-cli-extensions?view=azure-cli-latest&branch=pr-en-us-536) (azure-cli-latest) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C10%5C31%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201710311635563155-536%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=LmLJQo9%2F8VXvi%2BYihIO6qRHky0SmnDewaQ6YGnxmUzU%3D&st=2017-10-31T16%3A32%3A50Z&se=2017-12-01T16%3A37%3A50Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**